### PR TITLE
Admin changes status

### PR DESCRIPTION
**Pull request by simionescuraluca**

### CODE REVIEW

File: src/main/java/com/vinyl/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class OrderController {
+
+    @Autowired
+    OrderService orderService;
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/controller/OrderController.java]: private

File: src/main/java/com/vinyl/modelDTO/StatusDTO.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.vinyl.modelDTO;
+
+public class StatusDTO {
+
+    String status;
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/modelDTO/StatusDTO.java]: private

File: src/main/java/com/vinyl/controller/OrderController.java (hunk 1 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/controller/OrderController.java]: done

File: src/main/java/com/vinyl/modelDTO/StatusDTO.java (hunk 2 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/modelDTO/StatusDTO.java]: done

File: src/main/java/com/vinyl/service/OrderService.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    public void changeOrderStatus(StatusDTO newStatus, Integer orderId, String tokenHash){
+
+        validateAdminToken(tokenHash);
+
+        if(!newStatus.getStatus().equals("SENT")){
+            throw new BadRequestException("The order status is invalid!");
+        }
+
+        Optional order = purchaseRepository.findById(orderId);
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/OrderService.java]: Purchase p = purchaseRepository.findById(orderId).orElseThrow(new NotFoundException("The order is not found!"));

File: src/main/java/com/vinyl/service/OrderService.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        if(order == empty()){
+            throw new NotFoundException("The order is not found!");
+        }
+
+        Purchase newOrder = purchaseRepository.findById(orderId).get();
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/OrderService.java]: no need to find it from the db again

File: src/main/java/com/vinyl/service/OrderService.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        newOrder.setStatus(newStatus.getStatus());
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/service/OrderService.java]: only change the status if it is different

File: src/test/java/com/vinyl/ChangeOrderStatusTest.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    public void testWhenOk() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    }
+
+    @Test
+    public void testWhenStatusIsInvalid(){
+        status.setStatus("INVALID_STATUS");
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
```
Comment (dragosslujeru) [inline src/test/java/com/vinyl/ChangeOrderStatusTest.java:48]: assert that purchase status has not changed

File: src/test/java/com/vinyl/ChangeOrderStatusTest.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @Test
+    public void testWhenOrderIdIsInvalid(){
+        order.setId(00);
```
Comment (dragosslujeru) [inline src/test/java/com/vinyl/ChangeOrderStatusTest.java:54]: do the setUpHeaderAndGetTheResponse() overload and send the invalid id like that

File: src/main/java/com/vinyl/service/OrderService.java (hunk 3 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/OrderService.java]: done

File: src/main/java/com/vinyl/service/OrderService.java (hunk 4 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/OrderService.java]: done

File: src/main/java/com/vinyl/service/OrderService.java (hunk 5 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/service/OrderService.java]: I think it is always different; this endpoint focuses on changing the status of an order

File: src/test/java/com/vinyl/ChangeOrderStatusTest.java (hunk 6 above)
Comment (simionescuraluca) [reply src/test/java/com/vinyl/ChangeOrderStatusTest.java:48]: done

File: src/test/java/com/vinyl/ChangeOrderStatusTest.java (hunk 7 above)
Comment (simionescuraluca) [reply src/test/java/com/vinyl/ChangeOrderStatusTest.java:54]: this method calls the getUrl method which uses the id, so i should also parametrize the getUrl function to parametrize the setUp

**Review by dragosslujeru — APPROVED** [review]

**Pull request merged**